### ISSUE TITLE
Perform rendering in parallel

### ISSUE THREAD

**Issue by benreid24** [benreid24/BLIB#205]
With a dedicated thread and some re-work of how the renderer is invoked (currently systems but can be direct function calls) we can fully parallelize rendering. Care must be taken to avoid overlap, but the following schedules should work:

### Main thread
1. Process events
2. Perform game update logic
3. Wait signal `RenderDone`
4. Perform deferred ECS deletions
5. Signal `RenderStart`
6. Delay to maintain framerate
7. Wait signal `DescriptorSyncDone`
8. Loop

### Renderer thread
1. Wait signal `RenderStart`
2. Perform descriptor syncing
3. Signal `DescriptorSyncDone`
4. Record render commands
5. Signal `RenderDone`
6. Loop

There will likely be some extra synchronization issues not solved by the above schedule, for example Overlay rendering which is ECS based. In that specific case ECS transactions should be a viable solution.
